### DISCUSSION TITLE
Subscriptions with an arbitrary price

### DISCUSSION THREAD

**Discussion by brandonkelly** [craftcms/commerce#2332] [Ideas]
It should be possible to create a subscription with a customer-defined price. Especially useful for recurring donations ([example](https://givewp.com/addons/recurring-donations/)).

**Comment by KhaledGabr** [comment]
This is exactly what I am trying to do!  I need to allow subscription donations defined by user. I would image that this is possible with Adjusters, where I can set the product price to be 0 and and increase its price based on a value the the user selects in the front end. It would be really helpful if we can have a code sample for that in the documentation. I am not really sure, how can I do this.

**Comment by lukeholder** [comment]
We spoke to Stripe developers on the phone, and they said this is not possible unless you generate a plan on the fly for the user. Leaving this open for now.

**Comment by chasegiunta** [comment]
@brandonkelly @lukeholder I think the way other services _may_ be doing this is utilizing units on Product pricing plans:

(screenshots below are showing by $0.01, which I would think is rare. I'd think whole dollar amounts would be more common, but ultimately it would be up to the developer)

Pricing plan creation screen:

![image](https://user-images.githubusercontent.com/1377169/70382862-c6a04580-1928-11ea-9e5d-30c241a6e9c1.png)

-----

Stripe's subscription creation screen:
![image](https://user-images.githubusercontent.com/1377169/70382902-cf454b80-1929-11ea-847d-5ce97bbd2449.png)

If Commerce could tap into **units** on frontend forms, that would work. Not the most elegant solution to offer, but better than no support at all! And would probably be easier (and cleaner on the Stripe side) than dynamically generating plans.

**Comment by chasegiunta** [comment]
@lukeholder @brandonkelly I want follow up on the suggestion above, as we've had another client who is currently using Stripe to handle arbitrary subscription donations, but trying to move them over to Commerce, and we're unable to give them that functionality..

Seems like there's two ways to do this:

First way, which I've seen done and @lukeholder mentioned above, is dynamically creating the product plan on the fly for the user. Commerce isn't currently creating Stripe products so this would be a special use case, and it would _litter_ Stripe with single-customer-use Products. Not ideal.

The second way is how I've mentioned it above, and I believe it shouldn't be terrible to implement on the Commerce side, and that is being able to pass **_units_** for subscription payments.

Commerce is already tapping into Stripe's subscriptions, but it's only supporting the Standard pricing subscription model, not the Package pricing subscription model (correct me if I'm wrong).

**Comment by lukeholder** [comment]
@chasegiunta correct, we only support the basic fixed subscription price at the moment. We do want to improve subscriptions with better integration with price per unit, but it won't be coming in the short term. Happy to work with you on updates to the commerce gateway subscription API if you want to make a PR for this functionality.
